pkg/install/billing.go

```diff
@@ -21,10 +29,65 @@ func (i *Installer) createBillingRecord(ctx context.Context) error {
 			Location: i.doc.OpenShiftCluster.Location,
 		},
 	})
-	// If create return a conflict, this means row is already present in database
-	if err, ok := err.(*cosmosdb.Error); ok && err.StatusCode == http.StatusConflict {
-		return nil
+	if err != nil {
+		if err, ok := err.(*cosmosdb.Error); ok && err.StatusCode == http.StatusConflict {
+			i.log.Printf("billing record already present id DB")
+			return nil
+		}
+		return err
+	}
+	// Validate if E2E Feature is registred
+	resource, err := azure.ParseResourceID(billingDoc.Key)
+	if err != nil {
+		return err
 	}
 
-	return err
+	subDocument, err := i.sub.Get(ctx, resource.SubscriptionID)
+	if err != nil {
+		return err
+	}
+	if isSubscriptionRegisteredToE2E(*subDocument) {
+		// We are not failing the operation if we cannot write to e2e storage account, just warning
+		// The blob can only be created when the billing record is also written for the first time
+		errBlob := createE2Eblob(ctx, i.env, *billingDoc, resource)
+		if errBlob != nil {
+			i.log.Warnf("createE2Eblob failed : %s", errBlob.Error())
+		}
+	}
+	return nil
+}
+
+func createE2Eblob(ctx context.Context, env env.Interface, doc api.BillingDocument, resource azure.Resource) error {
+	client, err := azstorage.NewBasicClient(env.E2EStorageAccountName(), env.E2EStorageAccountKey())
+	if err != nil {
+		return err
+	}
+	bs := client.GetBlobService()
+	// Create container if it does not exist: [bill]-[region]-[resourceGroupName]-[resourceName]
+	containerName := strings.ToLower("bill-" + doc.Billing.Location + "-" + resource.ResourceGroup + "-" + resource.ResourceName)
+	if len(containerName) > 63 {
```

Comment:
`24[bill-australiasoutheast-]+[RG-Name]+[ResourceName]` Maybe we should validate that on the creation that `ResourceGroup+ResourceName` would not exceed this?

Reply:
@jim-minter

Reply:
I'm against restricting our users based on naming conventions for our test cases.  I'd be more for making our tests/test framework validate that the names it passes in won't exceed this number.